tests/ha/single_pod_restart/test_server_pod_restart.py

```diff
@@ -716,4 +717,106 @@ def test_reads_during_server_pod_restart(self):
+        LOGGER.info("Step 3: Start WRITEs with variable object sizes in background")
+        self.test_prefix_deg = 'test-44837-deg'
+        args = {'s3userinfo': list(users.values())[0], 'log_prefix': self.test_prefix_deg,
+                'skipread': True, 'skipcleanup': True, 'nclients': 1, 'nsamples': 30,
+                'setup_s3bench': False, 'output': output}
+        thread = threading.Thread(target=self.ha_obj.event_s3_operation,
+                                  args=(event,), kwargs=args)
+        thread.daemon = True  # Daemonize thread
+        thread.start()
+        LOGGER.info("Step 3: Started WRITEs with variable sizes objects in background.")
+        LOGGER.info("Waiting for %s seconds", HA_CFG["common_params"]["30sec_delay"])
+        time.sleep(HA_CFG["common_params"]["30sec_delay"])
+        LOGGER.info("Step 4: Restart the pod with replica method")
+        event.set()
+        resp = self.ha_obj.restore_pod(pod_obj=self.node_master_list[0],
+                                       restore_method=self.restore_method,
+                                       restore_params={"deployment_name": self.deployment_name,
+                                                       "deployment_backup":
+                                                           self.deployment_backup,
+                                                       "num_replica": self.num_replica,
+                                                       "set_name": self.set_name},
+                                       clstr_status=True)
+        LOGGER.debug("Response: %s", resp)
+        assert_utils.assert_true(resp[0], f"Failed to restore pod by {self.restore_method} way")
+        LOGGER.info("Step 4: Successfully restart the pod with replica method")
+        self.restore_pod = False
+        event.clear()
+        thread.join()
+        LOGGER.info("Step 5: Verifying writes from background process")
+        responses = dict()
+        while len(responses) != 2:
+            responses = output.get(timeout=HA_CFG["common_params"]["60sec_delay"])
+        pass_logs = list(x[1] for x in responses["pass_res"])
+        fail_logs = list(x[1] for x in responses["fail_res"])
+        LOGGER.debug("Pass logs list: %s \nFail logs list: %s", pass_logs, fail_logs)
+        resp = self.ha_obj.check_s3bench_log(file_paths=pass_logs)
+        assert_utils.assert_false(len(resp[1]), f"Logs which contain failures: {resp[1]}")
+        resp = self.ha_obj.check_s3bench_log(file_paths=fail_logs, pass_logs=False)
+        assert_utils.assert_true(len(resp[1]) <= len(fail_logs),
+                                 f"Logs which contain passed IOs: {resp[1]}")
+        LOGGER.info("Step 5: Successfully completed writes in background")
+        LOGGER.info("Step 6: Read/Verify data written in background process")
+        resp = self.ha_obj.ha_s3_workload_operation(s3userinfo=list(users.values())[0],
```

Comment:
Need to give sample value 30 as given in background call. Also here we can keep nclients to 30.